### DISCUSSION TITLE
Testnet DEX Now Available

### DISCUSSION THREAD

**Discussion by KoffeeBarista** [kcc-community/any-advice-issue#17] [Show and tell]
Hello everyone!

First off, just want to say how excited I am to be a part of this community. KuCoin is by far my favorite exchange and am looking forward to growing this network as much as possible.

We have launched a DEX on testnet so that people are able to deploy tokens, add liquidity, and swap them. We thought that getting something like this up as fast as possible was good for generating transactions and testing the network.

The DEX can currently be reached at: [https://test.koffeeswap.exchange](https://test.koffeeswap.exchange)

It's not pretty (yet), but it's functional. We will be updating it as regularly as possible and look forward to joining mainnet with everyone here.

Thanks!

p.s. If you run into any issues feel free to drop a comment here.

**Comment by viaweb3** [comment]
great work, bro

**Reply by viaweb3** [reply]
and you can join our developer group

**Reply by shoaibyazdani** [reply]
Share link if possible

**Comment by caizhiyuanrise** [comment]
Hello
How would I be able to play around with the testnet?
I mean I think I would not send my KCS from Kucoin over here, coreect? 
Thank you!

**Comment by shoaibyazdani** [comment]
Hy there, I need a testnet router adress, and plz help me i am trying to deploy a token having safemoon feature but i am unable to deploy facing some error like the router of Mainnet is not Valid or something else
